common/oauthTokenManager.go

```diff
@@ -713,15 +718,81 @@ func (credInfo *OAuthTokenInfo) GetNewTokenFromMSI(ctx context.Context) (*adal.T
 	if credInfo.IdentityInfo.MSIResID != "" {
 		params.Set("msi_res_id", credInfo.IdentityInfo.MSIResID)
 	}
+
 	req.URL.RawQuery = params.Encode()
 	req.Header.Set("Metadata", "true")
+
 	// Set context.
 	req.WithContext(ctx)
 
 	// Send request
 	resp, err := msiTokenHTTPClient.Do(req)
+
+	return req, resp, err
+}
+
+// Dated 15th Sep 2021.
+// Token JSON returned by ARC-server endpoint API currently does not set a valid integral value for "not_before" key.
+// Fix will be available in future ARC versions, till then use the current time to fixup the byte slice before we pass it to json.Unmarshal().
+// If the token JSON already has "not_before" correctly set, this will be a no-op.
+// fixupTokenJson corrects the value of JSON field "not_before" in the Byte slice from blank to a valid value and returns the corrected Byte slice.
```

Comment:
In accordance with godoc, have this as the very first line in the comment.
then leave a blank line and have rest of the comment lines.